activesupport/lib/active_support/core_ext/object/to_query.rb

```diff
@@ -81,4 +81,40 @@ def to_query(namespace = nil)
   end
 
   alias_method :to_param, :to_query
+
+  # Returns an array of hashes each containing :name and :value keys
+  # suitable for use as the names and values of form input fields:
+  #
+  #   {name: 'David', nationality: 'Danish'}.to_query
+  #   # => [{name: 'name', value: 'David'}, {name: 'nationality', value: 'Danish'}]
+  #
+  #   {country: {name: 'Denmark'}.to_query
+  #   # => [{name: 'country[name]', value: 'Denmark'}]
+  #
+  #   {countries: ['Denmark', 'Sweden']}.to_query
+  #   # => [{name: 'countries[]', value: 'Denmark'}, {name: 'countries[]', value: 'Sweden'}]
+  #
+  # An optional namespace can be passed to enclose key names:
+  #
+  #   {name: 'Denmark'}.to_query('country')
+  #   # => [{name: 'country[name]', value: 'Denmark'}]
+  def to_form_params(namespace = nil)
```

Comment:
This method is only used in one place. It probably doesn't need to live in Active Support for general use. IMO, it'd make more sense as a private method in `ActionView::Helpers::UrlHelper`.